File: examples/python/chat_completions.py
#!/usr/bin/env python3
"""OpenAI-compatible chat completions example for ViralAPI.

Environment variables:
  VIRALAPI_API_KEY   Your ViralAPI API key
  VIRALAPI_BASE_URL  Your OpenAI-compatible base URL, e.g. https://your-endpoint/v1
  VIRALAPI_MODEL     Optional model name configured for your account
"""

import json
import os
import sys
import urllib.request

api_key = os.environ.get("VIRALAPI_API_KEY")
base_url = os.environ.get("VIRALAPI_BASE_URL")
model = os.environ.get("VIRALAPI_MODEL", "gpt-4o-mini")

if not api_key or not base_url:
    sys.exit("Please set VIRALAPI_API_KEY and VIRALAPI_BASE_URL first.")

url = base_url.rstrip("/") + "/chat/completions"
payload = {
    "model": model,
    "messages": [
        {"role": "system", "content": "You are a concise API integration assistant."},
        {
            "role": "user",
            "content": "Give a short checklist for testing an OpenAI-compatible API gateway.",
        },
    ],
    "temperature": 0.3,
}

request = urllib.request.Request(
    url,
    data=json.dumps(payload).encode("utf-8"),
    headers={
        "Authorization": f"Bearer {api_key}",
        "Content-Type": "application/json",
    },
    method="POST",
)

with urllib.request.urlopen(request, timeout=60) as response:
    body = response.read().decode("utf-8")
    print(json.dumps(json.loads(body), ensure_ascii=False, indent=2))

File: examples/node/chat-completions.mjs
#!/usr/bin/env node
// OpenAI-compatible chat completions example for ViralAPI.
//
// Environment variables:
//   VIRALAPI_API_KEY   Your ViralAPI API key
//   VIRALAPI_BASE_URL  Your OpenAI-compatible base URL, e.g. https://your-endpoint/v1
//   VIRALAPI_MODEL     Optional model name configured for your account

const apiKey = process.env.VIRALAPI_API_KEY;
const baseUrl = process.env.VIRALAPI_BASE_URL;
const model = process.env.VIRALAPI_MODEL || "gpt-4o-mini";

if (!apiKey || !baseUrl) {
  console.error("Please set VIRALAPI_API_KEY and VIRALAPI_BASE_URL first.");
  process.exit(1);
}

const response = await fetch(`${baseUrl.replace(/\/$/, "")}/chat/completions`, {
  method: "POST",
  headers: {
    Authorization: `Bearer ${apiKey}`,
    "Content-Type": "application/json",
  },
  body: JSON.stringify({
    model,
    messages: [
      { role: "system", content: "You are a concise API integration assistant." },
      {
        role: "user",
        content:
          "Explain how a small team can switch models without rewriting app code.",
      },
    ],
    temperature: 0.3,
  }),
});

if (!response.ok) {
  console.error(`HTTP ${response.status}`);
  console.error(await response.text());
  process.exit(1);
}

console.log(JSON.stringify(await response.json(), null, 2));
